app/lib/lnd/walletUnlocker.js

```diff
@@ -115,6 +87,39 @@ class WalletUnlocker {
   // Helpers
   // ------------------------------------
 
+  /**
+   * Establish a connection to the Lightning interface.
+   */
+  async establishConnection() {
+    const { host, cert } = this.lndConfig
+
+    // Find the most recent rpc.proto file
+    const version = await getLatestProtoVersion()
+    const protoFile = await getProtoFile(version)
+    mainLog.debug('Establishing gRPC connection with proto file %s', protoFile)
+
+    // Load gRPC package definition as a gRPC object hierarchy.
+    const packageDefinition = await load(protoFile, grpcOptions)
+    const rpc = loadPackageDefinition(packageDefinition)
+
+    // Create ssl credentials to use with the gRPC client.
+    const sslCreds = await createSslCreds(cert)
+
+    // Create a new gRPC client instance.
+    this.service = new rpc.lnrpc.WalletUnlocker(host, sslCreds)
+
+    // Wait upto 20 seconds for the gRPC connection to be established.
+    return new Promise((resolve, reject) => {
```

Comment:
Looks like this can be promisified to avoid `new Promise`

Reply:
This PR is off master which doesn't have the new `promisifiedCall` method. Given that the PR to add that was fairly large and we want to keep changes to a minimum in master, I'd prefer to leave this as it is for now.

Note, I didn't write this method in the PR - I have just pulled it out from another method where it was previously embedded to make the code a little clearer.